tests/fuzzer.rs

```diff
@@ -0,0 +1,156 @@
+use flowgger;
+use quickcheck;
+
+use quickcheck::QuickCheck;
+
+use flowgger::flowgger::config::Config;
+use flowgger::flowgger::encoder::Encoder;
+use flowgger::flowgger::decoder::Decoder;
+use flowgger::flowgger::merger;
+use flowgger::flowgger::output;
+
+use std::sync::mpsc::{Receiver, sync_channel, SyncSender};
+
+use flowgger::flowgger::get_decoder_rfc3164;
+use flowgger::flowgger::get_encoder_rfc3164;
+use flowgger::flowgger::input::udp_input::handle_record_maybe_compressed;
+
+use self::merger::{LineMerger, Merger};
+use self::output::FileOutput;
+use self::output::Output;
+
+use std::sync::{Arc, Mutex};
+use toml::Value;
+use std::fs;
+use std::{thread, time};
+
+const DEFAULT_CONFIG_FILE: &str = "flowgger.toml";
+const DEFAULT_OUTPUT_FILEPATH: &str = "output.log";
+const DEFAULT_QUEUE_SIZE: usize = 10_000_000;
+
+const DEFAULT_OUTPUT_FORMAT: &str = "gelf";
+const DEFAULT_OUTPUT_FRAMING: &str = "noop";
+const DEFAULT_OUTPUT_TYPE: &str = "file";
+
+const DEFAULT_FUZZED_MESSAGE_COUNT: u64 = 500;
+
+fn get_file_output(config: &Config) -> Box<dyn Output> {
+    Box::new(FileOutput::new(config)) as Box<dyn Output>
+}
+
+pub fn start_file_output(config: &Config, rx: Receiver<Vec<u8>>){
+
+    let output_format = config
+        .lookup("output.format")
+        .map_or(DEFAULT_OUTPUT_FORMAT, |x| {
+            x.as_str().expect("output.format must be a string")
+        });
+
+    let output = get_file_output(&config);
+    let output_type = config
+        .lookup("output.type")
+        .map_or(DEFAULT_OUTPUT_TYPE, |x| {
+            x.as_str().expect("output.type must be a string")
+        });
+
+    let _output_framing = match config.lookup("output.framing") {
+        Some(framing) => framing.as_str().expect("output.framing must be a string"),
+        None => match (output_format, output_type) {
+            ("capnp", _) | (_, "kafka") => "noop",
+            (_, "debug") | ("ltsv", _) => "line",
+            ("gelf", _) => "nul",
+            _ => DEFAULT_OUTPUT_FRAMING,
+        },
+    };
+    let merger: Option<Box<dyn Merger>> = Some(Box::new(LineMerger::new(&config)) as Box<dyn Merger>);
+
+    let arx = Arc::new(Mutex::new(rx));
+    output.start(arx, merger);
+
+}
+
+pub fn get_config() -> Config {
+    let mut config = match Config::from_path(DEFAULT_CONFIG_FILE) {
+        Ok(config) => config,
+        Err(e) => panic!(
+            "Unable to read the config file [{}]: {}",
+            "flowgger.toml",
+            e.to_string()
+        ),
+    };
+
+    if let Some(entry) = config.config.get_mut("output").unwrap().get_mut("file_rotation_time"){
+        *entry = Value::Integer(0);
+    }else{
+        panic!("Failed to find config entry");
+    }
+
+    return config;
+}
+
+pub fn remove_output_file(file_output_path: &str){
+    fs::remove_file(file_output_path);
+}
+
+pub fn fuzz_target_rfc3164(data: &[u8]) {
```

Comment:
so, am i getting this right ?are you creating a listener for each message, send the message, check the output file, delete it, close the listeners, and do that again ? 500 times ???

why not just create the listeners, fuzz data with ` handle_record_maybe_compressed` and open the output file just once, parse the records, and error out if one can't be read ?

Reply:
addressed in next rev